controller/konnect/index_kongpluginbinding.go

```diff
@@ -52,3 +80,55 @@ func kongClusterPluginReferencesFromKongPluginBinding(obj client.Object) []strin
 	}
 	return []string{binding.Spec.PluginReference.Name}
 }
+
+// kongServiceReferencesFromKongPluginBinding returns name of referenced KongService in KongPluginBinding spec.
+func kongServiceReferencesFromKongPluginBinding(obj client.Object) []string {
+	binding, ok := obj.(*configurationv1alpha1.KongPluginBinding)
+	if !ok {
+		return nil
+	}
+	if binding.Spec.Targets.ServiceReference == nil ||
```

Comment:
We did not include the namespace in the index key while `KongService`,`KongRoute`,`KongConsumer` and `KongConsumerGroup` are namespaced. Would there be conflicts when different resources are using a same name in different namespaces, thus unnecessary reconciliations are triggered?

Reply:
This is a good point, I've added the namespace filter on all the watchers 👍